librad/src/meta/entity.rs

```diff
@@ -502,11 +459,11 @@ where
     /// - the entity has not been already signed using this same key
     /// - this key is allowed to sign the entity (using `check_key`)
     pub async fn sign(
-        &mut self,
+        mut self,
         key: &Key,
         by: &Signatory,
-        resolver: &impl Resolver<User<EntityStatusUnknown>>,
-    ) -> Result<(), Error> {
+        resolver: &impl Resolver<User<Unknown>>,
+    ) -> Result<Entity<T, Signed>, Error> {
```

Comment:
This should not be a signed entity because invoking `sign` signs with just one key.
There's no guarantee that the entity has been signed with *all* keys.
And returning an error here makes no sense: the entity has been signed with the provided key, the operation is successful, it's just that the entity is not necessarily _completely_ signed.

Reply:
Ah great point! :smile: Thanks for catching that